designs/cluster-status.md

```diff
@@ -0,0 +1,406 @@
+    ObservedGeneration int64`json:"observedGeneration,omitempty"`
+  }
+```
+
+```
+// ClusterCNI represents a CNI cluster component.
+type ClusterCNI struct {
+    // E.g. cilium
+    Name string`json:"name,omitempty"`
+    
+    // Empty if not applied or no version can be inferred.
+    // +optional
+    Version string`json:"version,omitempty"`
+    
+    // Applied, Not Applied
+    Status string`json:"status,omitempty"`
+}
+```
+
+**Default CNI field**
+
+EKS Anywhere deploys EKS Anywhere Cilium as the default managed CNI option. However, this design is informed by the fact that we do support other CNI solutions like `kindnetd`. Although, this is only for docker and thus, non-production clusters, EKS Anywhere has the capability for configuring managed CNI solutions beyond cilium and can possibly support more in the future.
+
+The user may need information on managed CNI component in the `Cluster` status for a few reasons. First, the name and version of CNI component installed may serve as initial point for debugging related networking issues. Users may also opt to use a custom CNI instead of the default we use in EKS Anywhere (Cilium), where at first Cilium is installed on cluster creations because it’s necessary for provisioning. However, after the user would have to manually uninstall  EKS Anywhere Cilium and replace it with a different CNI. In this case, it would be useful for the user to be able to verify the managed CNI is installed or not. 
+
+**Adding new conditions**
+
+Conditions provide a mechanism for high-level status reporting from the controller and a collection of conditions can be used represent a more specific assessment of cluster readiness. We should use multiple conditions to assessment  the user a the status of the individual checkpoints defined above and then use the conditions to determine overall cluster readiness. This gives the user insight into the checkpoint states of cluster readiness that we’ve defined, and allows other components to avoid duplicating the logic of each condition in the controller. Keeping the information that we need to collect that we’ve defined above in mind, we propose adding the following condition types:
+
+
+* `ControlPlaneInitialized` - reports the first control plane has been initialized and the cluster’s API server is contactable via the available kubeconfig generated in the management cluster as a secret. Once this condition is marked true, it’s value never changes.
+* `ControlPlaneReady` -  reports that the condition of the current state of the specified control plane machines vs the desired state in the Cluster spec. e.g. replicasReady vs replicas,
+* `DefaultCNIConfigured` - reports that the CNI has been successfully configured. If an EKS Anywhere managed CNI solution is not configured and the `Cluster` spec is configured to skip EKS Anywhere's default CNI upgrades, this condition will be marked as “False” with the reason “UsingCustomCNI”.
```

Comment:
Nit: Should we call this `DefaultCNIInstalled`? Are we doing anything less than that to just "configure" it?

Reply:
I preferred `DefaultCNIConfigured` because an EKSA cluster can also be configured to skip EKS Anywhere's default CNI upgrades and we'll also use this status to describe that